olog.py

```diff
@@ -16,77 +15,221 @@ def __init__(self, url, user, password):
-        self._session.verify = False
-        self._kwargs = dict(headers=headers, auth=auth)  # for every request
-
-    def list_logbooks(self):
-        url = f'{self._url}/resources/logbooks'
-        res = self._session.get(url, **self._kwargs)
+        if not url.endswith('/'):
+            url += '/'
+        self._session = AsyncClient(base_url=url, auth=(user, password))
+
+    # Logbooks
+    async def aget_logbooks(self):
+        async with self._session as api:
+            res = await api.get('logbooks')
         res.raise_for_status()
         return res.json()
 
-    def get_logbook(self, name):
+    def get_logbooks(self):
+        return asyncio.run(self.aget_logbooks())
+
+    async def aget_logbook(self, name):
         # Logbooks have an integer id, but this REST endpoint expects the
         # *name*.  It does not accept an id.
-        url = f'{self._url}/resources/logbooks/{name}'
-        res = self._session.get(url, **self._kwargs)
+        async with self._session as api:
+            res = await api.get(f'logbooks/{name}')
         res.raise_for_status()
         return res.json()
 
-    def delete_logbook(self, name):
-        # Logbooks have an integer id, but this REST endpoint expects the
-        # *name*.  It does not accept an id.
-        url = f'{self._url}/resources/logbooks/{name}'
-        res = self._session.delete(url, **self._kwargs)
+    def get_logbook(self, name):
+        return asyncio.run(self.aget_logbook(name))
+
+    async def aput_logbooks(self, logbooks):
+        async with self._session as api:
+            res = await api.put('logbooks', json=logbooks)
+        res.raise_for_status()
+        return None
+
+    def put_logbooks(self, logbooks):
+        return asyncio.run(self.aput_logbooks(logbooks))
+
+    async def aput_logbook(self, logbook):
+        async with self._session as api:
+            res = await api.put(f'logbooks/{logbook["name"]}',  json=logbook)
+            logbook_from_server = res.json()
+            logbook_from_server.pop('owner')
+            logbook.pop('owner')
+        if (not res.raise_for_status()) and (logbook != logbook_from_server):
+            raise ValueError("No http error was raised but server doesn't \
+                             successfully put logbook you want")
+        return None
+
+    def put_logbook(self, logbook):
+        return asyncio.run(self.aput_logbook(logbook))
+
+    # Logs
+    def ensure_time(self, input):
+        if isinstance(input, datetime):
+            return input.isoformat(sep=' ', timespec='milliseconds')
+        else:
+            d = datetime.strptime(input, '%Y-%m-%d %H:%M:%S.%f')
+            return d.isoformat(sep=' ', timespec='milliseconds')
+
+    async def aget_logs(self, desc=None, fuzzy=None, phrase=None, owner=None,
+                        start=None, end=None, includeevents=None,
+                        logbooks=None, tags=None, properties=None):
+        if start:
```

Comment:
This should be `if start is not None` because `start=0` is a valid input and would be `False` here. Same for `end` below.